src/SW_Weather.c

```diff
@@ -781,6 +867,108 @@ void generateMissingWeather(
   }
 }
 
+/**
+ @brief Check weather through all years/days within simultation and make sure all input values are reasonable
+ after possible weather generation and scaling. If a value is to be found unreasonable, the function will execute a program crash.
+
+ @param weather Struct of type SW_WEATHER holding all relevant information pretaining to weather input data
+ */
+void checkAllWeather(SW_WEATHER *weather) {
+
+    // Initialize any variables
+    TimeInt year, doy, numDaysInYear;
+    SW_WEATHER_HIST **weathHist = weather->allHist;
+
+    double dailyMinTemp, dailyMaxTemp;
+
+    // Check if minimum or maximum temperature, or precipitation flags are 0
+    if(!weather->has_temp2 || !weather->has_ppt) {

```

Comment:
Note sure that `checkAllWeather()` is the best location to fail if `!weather->has_temp2 || !weather->has_ppt` -- we want to enforce that on the input files, i.e., maybe `SW_WTH_read()` or `readAllWeather()` seem places that more directly deal with input files